bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "''",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "paired and nested with non-bracket chars",
+      "input": "([{f00: bar}({}[baz])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested with non-bracket chars",
+      "input": "([{this: sh0ld({}[be] false)])",
```

Comment:
I've just want to add some numbers to test.
May be I need to write separate test with only numbers or rewrite test to be more obvious.